### DISCUSSION TITLE
Coordinator Backup and restore

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy#1152] [Q&A] [answered]
In the case of EZSP , when doing the restore, I think that a special option has to be used in order to flash the Coordinator Mac@.
Is that something which is available as part of the zigpy/backup ?

Or is that something we need to do upfront via zigpy-cli or an other too ?

**Comment by puddly** [comment, marked as answer]
Set this to overwrite on restore:

```Python
backup.network_info.stack_specific.setdefault("ezsp", {})[
    "i_understand_i_can_update_eui64_only_once_and_i_still_want_to_do_it"
] = True
```

**Reply by pipiche38** [reply]
this has to be set in the backup to be restore ?
In other words, we have to update the coordinator backup and add this to the 'ezsp': {}

**Reply by puddly** [reply]
Yes. You can modify the object without modifying the backup itself.

**Comment by Hedda** [comment]
Note! If I remember correctly, on Silicon Labs SoCs you are only able to override EUI64 once via the serial interface, after that time? And after that, you will need to re-flash the firmware via JTAG/SWD (such as with a J-Link Debug Probe to flash via SWD) to reset the number of times it has registered that the EUI64 has been overridden. 

https://github.com/zigpy/bellows#nvram-backup-and-restore

So while the end user can restore the backup as many times as they want, they are limited to only overriding the EUI64 just once?

So might want to consider adding an extra warning like this if and when user try to override EUI64:

"_This backup can be restored on another Zigbee Coordinator hardware that uses the same SoC-chip type. If restoration is done between different types Zigbee Coordinator hardware then be careful overriding the EUI-64 (Extended Unique Identifier, e.i. the unique MAC address of the controller), as some types of controllers, like those based on Silicon Labs only accept a single EUI64 update via its EZSP serial interface. Any more than that and you will need to manually re-flash firmware via a JTAG/SWD debug probe adapter (such as those from J-Link)_".

**Reply by MattWestb** [reply]
> you will need to re-flash the firmware

That wrong then its not the counter that is flashed its the token with IEEE and its not in the main flash that being erased with "erase chip" or flashing new firmware that is erasing flash aria 0 then its stored in flash aria 2 "!user data" that is blocked in the bootloader for updates.

Best way is dumping the user data and the ding one chip recovery or debug unlock for getting the aria blanked. Also possible patching the dump and deleting the IEEE token and writing it back to aria 1.

PS i have not found one way flashing information with GBL files to aria 1 then its looks like Silabs commander is blocked adding files to it and can only being done with SWD commands after the chip left  the factory.
